labman/db/study.py

```diff
@@ -64,6 +67,86 @@ def creator(self):
         """The user that created the study"""
         return user.User(self._get_attr('email'))
 
+    def generate_sample_plate_maps(self):
+        """Generates a string with the plate maps in a csv format
+
+        Returns
+        -------
+        str
+        """
+        with sql_connection.TRN as TRN:
+            # Retrieve all the plate layouts for all the sample plates in which
+            # this study has been plated.
+            sql = """SELECT plate_id, external_id AS plate_name,
+                            num_rows, num_columns,
+                            array_agg((content, row_num, col_num)
+                                      ORDER BY row_num, col_num) AS layout
+                     FROM qiita.well
+                        JOIN qiita.composition USING (container_id)
+                        JOIN qiita.sample_composition USING (composition_id)
+                        JOIN qiita.plate USING (plate_id)
+                        JOIN qiita.plate_configuration
+                            USING (plate_configuration_id)
+                     WHERE plate_id IN (SELECT DISTINCT plate_id
+                                        FROM qiita.study_sample
+                                            JOIN qiita.sample_composition
+                                                USING (sample_id)
+                                            JOIN qiita.composition
+                                                USING (composition_id)
+                                            JOIN qiita.well
+                                                USING (container_id)
+                                        WHERE study_id = %s)
+                     GROUP BY plate_id, num_rows, num_columns
+                     ORDER BY plate_id"""
+            TRN.add(sql, [self.id])
+            res = TRN.execute_fetchindex()
+            # Create a list of numpy arrays to store the layouts
+            layouts = [np.zeros((nr, nc), dtype=object)
+                       for _, _, nr, nc, _ in res]
+            # Loop through all the plates and populate the layouts with the
+            # contents of the well
+            for idx, (_, _, _, _, layout) in enumerate(res):
+                # Description of the string mangling: The `array_agg` operation
```

Comment:
Thanks for adding this description, would have been pretty hard to follow otherwise. If you still think eval might be a way to go (security people might cringe about that), perhaps replacing the curly braces with at the beginning and end and replacing them for open and close squared braces would do the trick. Alternatively, if this is something that's been needed elsewhere it might also make sense to leave this parsing code as is but centralize it in a function outside of this method. That way the function can be properly tested for edge cases, documented, etc.